gef-ui/server/controllers/about-facility/index.test.js

```diff
@@ -582,6 +582,40 @@ describe('controllers/about-facility', () => {
       }));
     });
 
+    it('should show error message if cover starts on submission which is the same as the coverEndDate', async () => {
+      mockRequest.body.facilityType = CONSTANTS.FACILITY_TYPE.CASH;
+      mockRequest.body.hasBeenIssued = 'true';
+      mockRequest.body.shouldCoverStartOnSubmission = 'true';
+      mockRequest.body['cover-end-date-day'] = format(now, 'd');
+      mockRequest.body['cover-end-date-month'] = format(now, 'M');
+      mockRequest.body['cover-end-date-year'] = format(now, 'yyyy');
+
+      await validateAboutFacility(mockRequest, mockResponse);
+
+      expect(mockResponse.render).toHaveBeenCalledWith('partials/about-facility.njk', expect.objectContaining({
+        errors: expect.objectContaining({
+          errorSummary: expect.arrayContaining([{ href: '#coverEndDate', text: expect.any(String) }]),
+        }),
+      }));
+    });
+
+    it('should show error message if cover starts on submission which is before the coverEndDate', async () => {
+      mockRequest.body.facilityType = CONSTANTS.FACILITY_TYPE.CASH;
+      mockRequest.body.hasBeenIssued = 'true';
+      mockRequest.body.shouldCoverStartOnSubmission = 'true';
+      mockRequest.body['cover-end-date-day'] = format(yesterday, 'd');
+      mockRequest.body['cover-end-date-month'] = format(yesterday, 'M');
+      mockRequest.body['cover-end-date-year'] = format(yesterday, 'yyyy');
+
+      await validateAboutFacility(mockRequest, mockResponse);
+
+      expect(mockResponse.render).toHaveBeenCalledWith('partials/about-facility.njk', expect.objectContaining({
+        errors: expect.objectContaining({
+          errorSummary: expect.arrayContaining([{ href: '#coverEndDate', text: expect.any(String) }]),
```

Comment:
Can we please have exact textual error message instead of `expect.any(String)` to increase E2E reliability.

Reply:
done